icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java

```diff
@@ -161,13 +189,37 @@ private static String getPerUnitFormat(ULocale locale, UnitWidth width) {
         }
     }
 
+    private static String getDeriveCompoundRule(ULocale locale, String feature, String structure) {
+        ICUResourceBundle derivationsBundle =
+                (ICUResourceBundle) UResourceBundle
+                        .getBundleInstance(ICUData.ICU_BASE_NAME, "grammaticalFeatures");
+
+        derivationsBundle = (ICUResourceBundle) derivationsBundle.get("grammaticalData");
+        derivationsBundle = (ICUResourceBundle) derivationsBundle.get("derivations");
+
+        // TODO: use standard normal locale resolution algorithms rather than just grabbing language:
+        ICUResourceBundle stackBundle;
+        try {
+            stackBundle = (ICUResourceBundle) derivationsBundle.get(locale.getLanguage());
```

Comment:
```suggestion
        ICUResourceBundle stackBundle;
        try {
            // TODO: use standard normal locale resolution algorithms rather than just grabbing language:
            stackBundle = (ICUResourceBundle) derivationsBundle.get(locale.getLanguage());
```

Reply:
Fallback to "root" is indeed related to the fallback algorithm. I'm making this suggestion though to highlight the "getLanguage()" call, which is where we'll end up failing to pick up any regional variations (which don't currently exist, but might in the future).

Reply:
done